core/services/relay/evm/mercury/wsrpc/cache/cache_set.go

```diff
@@ -0,0 +1,118 @@
+package cache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+// NOTE: CacheSet holds a set of mercury caches keyed by server URL
+
+type CacheSet interface {
+	services.Service
+	Get(ctx context.Context, client Client) (Fetcher, error)
+}
+
+var _ CacheSet = (*cacheSet)(nil)
+
+type cacheSet struct {
+	sync.RWMutex
+	services.StateMachine
+
+	lggr   logger.Logger
+	caches map[string]Cache
+
+	latestPriceTTL time.Duration
+	maxStaleAge    time.Duration
+}
+
+func NewCacheSet(cfg Config) CacheSet {
+	return newCacheSet(cfg)
+}
+
+func newCacheSet(cfg Config) *cacheSet {
+	return &cacheSet{
+		sync.RWMutex{},
+		services.StateMachine{},
+		cfg.Logger.Named("CacheSet"),
+		make(map[string]Cache),
+		cfg.LatestReportTTL,
+		cfg.MaxStaleAge,
+	}
+}
+
+func (cs *cacheSet) Start(context.Context) error {
+	return cs.StartOnce("CacheSet", func() error {
+		return nil
+	})
+}
+
+func (cs *cacheSet) Close() error {
+	return cs.StopOnce("CacheSet", func() (merr error) {
+		cs.Lock()
+		defer cs.Unlock()
+		for _, c := range cs.caches {
+			merr = errors.Join(merr, c.Close())
+		}
```

Comment:
You could use a `MultiCloser` to close them in parallel:
```suggestion
		merr = services.MultiCloser(cs.caches).Close()
```

Reply:
It's a map, not an array. Can fanout the close tho

Reply:
Ah, well `Map.Values` then